pkg/resources/user.go

```diff
@@ -677,6 +679,40 @@ func GetUpdateUserFunc(userType sdk.UserType) func(ctx context.Context, d *schem
 	}
 }
 
+// handlePasswordUpdate is a current workaround to handle user's password after import.
+// Password is empty after the import, we can't read it from the config or from Snowflake.
+// During the next terraform plan+apply it's updated to the "same" value.
+// It results in an error on Snowflake side: New password rejected by current password policy. Reason: 'PRIOR_USE'.
+// Current workaround is to ignore such an error. We will revisit it after migration to plugin framework.
+func handlePasswordUpdate(ctx context.Context, id sdk.AccountObjectIdentifier, userType sdk.UserType, d *schema.ResourceData, client *sdk.Client) error {
+	if userType == sdk.UserTypePerson || userType == sdk.UserTypeLegacyService {
+		setPassword := sdk.UserAlterObjectProperties{}
+		unsetPassword := sdk.UserObjectPropertiesUnset{}
+		if err := stringAttributeUpdate(d, "password", &setPassword.Password, &unsetPassword.Password); err != nil {
+			return err
+		}
+		if (setPassword != sdk.UserAlterObjectProperties{}) {
+			err := client.Users.Alter(ctx, id, &sdk.AlterUserOptions{Set: &sdk.UserSet{ObjectProperties: &setPassword}})
+			if err != nil {
+				if strings.Contains(err.Error(), "Error: 003002 (28P01)") || strings.Contains(err.Error(), "Reason: 'PRIOR_USE'") {
```

Comment:
I think this should be handled on SDK level, and we could return a custom error from there.

Reply:
Okay, I can migrate it with the next PR